<a href="https://colab.research.google.com/github/yovngburgos/examenEstadistica/blob/main/Laboratorio_2_Tablas_de_frecuencia.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Tablas de frecuencias

## Objetivos:

Aplicar el análisis de frecuencias en distintas variables según su clasificación e interpretar los resultados.

*Nota: Este laboratorio está diseñado para desarrollar razonamiento estadístico. Evita el uso de herramientas de IA para responder las preguntas a menos que esto sea explícitamente sugerido* 🤖. *En su lugar, enfócate en aplicar los conceptos revisados.*

## Actividad 1

Netflix es una de las plataformas de streaming más populares del mundo. En esta actividad, trabajarás con una base de datos real de títulos disponibles en Netflix. El objetivo es explorar algunas características del contenido y observar qué tan comunes son ciertos tipos de programas o películas.

Descarga la base de datos <a href="https://docs.google.com/spreadsheets/d/1E9w_qc7U09eaEcybEw08SOB5-JuIvSlM/export?format=xlsx">aquí</a> y responde las siguientes preguntas:

In [ ]:
import pandas as pd

#Para cargar archivo excel en Python
excel_file = pd.ExcelFile('Base_Netflix.xlsx')
sheet_names = excel_file.sheet_names
print(sheet_names)

['Hoja1']


In [ ]:
base_datos=pd.read_excel("Base_Netflix.xlsx",sheet_name="Hoja1")

In [ ]:
base_datos.head(10)

ID      Tipo             Título                                   País  \
0   s47  Película         Safe House     South Africa, United States, Japan   
1   s82  Película               Kate                          United States   
2  s136  Película        Cliffhanger    United States, Italy, France, Japan   
3  s142  Película         Extraction  United States, United Kingdom, Canada   
4  s229  Película   The November Man          United States, United Kingdom   
5  s271  Película            Beckett                  Italy, Brazil, Greece   
6  s329  Película  Boyka: Undisputed                Bulgaria, United States   
7  s357  Película         The Losers                  United States, France   
8  s438  Película         Cosmic Sin                          United States   
9  s559  Película          6 Bullets                          United States   

    Año Duración          Categoría  
0  2012  115 min  Acción y Aventura  
1  2021  106 min  Acción y Aventura  
2  1993  113 min  Acción y Aventura  
3  2015   82 min  Acción y Aventura  
4  2014  108 min  Acción y Aventura  
5  2021  110 min  Acción y Aventura  
6  2016   90 min  Acción y Aventura  
7  2010   97 min  Acción y Aventura  
8  2021   88 min  Acción y Aventura  
9  2012  115 min  Acción y Aventura

1. Defina la población y muestra.

Población: Catálogo de Netflix

Muestra: 7095 elementos del catálogo de Netflix

2. ¿Qué clasificación tiene la variable *Tipo*?

In [ ]:
base_datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7095 entries, 0 to 7094
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   ID         7095 non-null   object
 1   Tipo       7095 non-null   object
 2   Título     7095 non-null   object
 3   País       7095 non-null   object
 4   Año        7095 non-null   int64 
 5   Duración   7095 non-null   object
 6   Categoría  7095 non-null   object
dtypes: int64(1), object(6)
memory usage: 388.1+ KB


In [ ]:
base_datos["Tipo"].unique()

array(['Película', 'TV Show'], dtype=object)

**Respuesta**: La variable "Tipo" es cualitativa nominal

3. ¿Qué tipo de contenido es más frecuente en la plataforma: *Película* o *TV show* (series)?

In [ ]:
tamaño=base_datos.shape[0]
cuenta_variable = base_datos.groupby('Tipo').size()
tabla = pd.DataFrame({
    'Frecuencia absoluta': cuenta_variable,
    'Frecuencia relativa (%)': round(cuenta_variable / tamaño * 100, 1)
})
tabla

Frecuencia absoluta  Frecuencia relativa (%)
Tipo                                                  
Película                 5673                     80.0
TV Show                  1422                     20.0

**Respuesta**: Las Películas son el contenido más frecuente

4. ¿Qué clasificación tiene la variable *Categoría*?

In [ ]:
base_datos["Categoría"].unique()

array(['Acción y Aventura', 'Anime', 'Películas familiares', 'Clásicos',
       'Comedia', 'Crimen', 'Documental', 'Docuseries', 'Dramas',
       'Horror', 'Películas', 'Niños', 'Musical', 'Reality Show',
       'Romántico', 'Ciencia Ficción', 'Stand-Up Comedy', 'Thrillers'],
      dtype=object)

**Respuesta**: La variable "Categoría" es cualitativa nominal

5. ¿Cuál es la categoría de contenido más común? ¿A qué crees que se debe esto?

In [ ]:
cuenta_variable = base_datos.groupby('Categoría').size()
tabla2 = pd.DataFrame({
    'Frecuencia absoluta': cuenta_variable,
    'Frecuencia relativa (%)': round(cuenta_variable / tamaño * 100, 1)
})
tabla2.sort_values('Frecuencia absoluta', ascending= False)

Frecuencia absoluta  Frecuencia relativa (%)
Categoría                                                         
Dramas                               1593                     22.5
Comedia                              1266                     17.8
Acción y Aventura                     856                     12.1
Documental                            759                     10.7
Películas familiares                  500                      7.0
Crimen                                358                      5.0
Stand-Up Comedy                       334                      4.7
Niños                                 318                      4.5
Horror                                273                      3.8
Docuseries                            175                      2.5
Anime                                 173                      2.4
Películas                             165                      2.3
Clásicos                              100                      1.4
Reality Show                           99                      1.4
Thrillers                              62                      0.9
Romántico                              35                      0.5
Musical                                15                      0.2
Ciencia Ficción                        14                      0.2

**Respuesta:** La categoría más frecuente es "Drama".

6. Crea una tabla que contenga las frecuencias absolutas y relativas de la cantidad de títulos por categoría. Luego, interpreta la frecuencia relativa de las categorías *Documental* y *Comedia*, y responde ¿a qué razón atribuyes la diferencia entre los valores de las frecuencias?

**Respuesta:** Esto ya fue creado en el punto 5.

Documental=10,7%
Interpretación: El 10,7% de los títulos corresponden a documentales.

Comedia=17,8%
Interpretación: El 17,8% de los títulos corresponden a comedia

7. Construya una tabla de frecuencias absolutas y relativas de los 10 países con mayor cantidad de títulos, ¿es uniforme la presencia en la plataforma de estos 10 países? ¿Por qué piensas que ocurre lo anterior?

In [ ]:
cuenta_variable = base_datos.groupby('País').size()
tabla2 = pd.DataFrame({
    'Frecuencia absoluta': cuenta_variable,
    'Frecuencia relativa (%)': round(cuenta_variable / tamaño * 100, 1)
})
tabla2.sort_values('Frecuencia absoluta', ascending= False).head(10)

Frecuencia absoluta  Frecuencia relativa (%)
País                                                        
United States                  2779                     39.2
India                           918                     12.9
United Kingdom                  221                      3.1
Japan                           213                      3.0
Canada                          166                      2.3
Spain                           120                      1.7
France                          106                      1.5
Mexico                           96                      1.4
Egypt                            94                      1.3
Nigeria                          87                      1.2

## Actividad 2

La siguiente base de datos contiene información de estudiantes pertenecientes a un Instituto Profesional. Con el fin de tomar decisiones institucionales, como reforzar asignaturas o rediseñar horarios se registraron datos sobre su rendimiento académico, género, jornada y número de asignaturas reprobadas.

Para responder las siguientes preguntas, descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1Z2mrhxQnGgEfMqKrPyXBBPLnFPg52Woz
">aquí</a>**.

In [ ]:
base_datos = pd.read_csv('Calificaciones.csv')
base_datos

id     Género     Jornada Asignaturas reprobadas Semestre ingreso
0       1   Femenino       Diuna                      0          Primero
1       2   Femenino       Diuna                      3          Primero
2       3  Masculino       Diuna                      4          Primero
3       4  Masculino       Diuna                      1          Segundo
4       5   Femenino       Diuna                      3          Primero
..    ...        ...         ...                    ...              ...
995   996  Masculino       Diuna                      3          Segundo
996   997   Femenino  Vespertina                      3          Primero
997   998   Femenino  Vespertina                      1          Segundo
998   999  Masculino       Diuna                      1          Segundo
999  1000  Masculino  Vespertina                      2          Segundo

[1000 rows x 5 columns]

In [ ]:
tamaño=base_datos.shape[0]
print(tamaño)

1000


1. Define la población y la muestra del estudio.

**Respuesta**:

Población: Estudiantes de un instituto profesional

Muestra: 1000 estudiantes de un instituto profesional

2. ¿Cuál es la clasificación de las variables presentes en el estudio?

In [ ]:
base_datos["Género"].unique()

array(['Femenino', 'Masculino'], dtype=object)

In [ ]:
base_datos["Jornada"].unique()

array(['Diuna', 'Vespertina'], dtype=object)

In [ ]:
base_datos["Asignaturas reprobadas"].unique()

array(['0', '3', '4', '1', '2', '5 o más'], dtype=object)

In [ ]:
base_datos["Semestre ingreso"].unique()

array(['Primero', 'Segundo'], dtype=object)

**Respuesta:**

Género: Cualitativa nominal

Jornada: Cualitativa nominal

Asignaturas reprobadas: Cualitativa ordinal

Semestre ingreso: Cualitativa ordinal

3. ¿Cuál es el género más común entre los estudiantes? ¿Qué porcentaje representa?

In [ ]:
#Tamaño de la base de datos
tamaño=base_datos.shape[0]

#Tabla de frecuencia variable cualitativa
cuenta_variable = base_datos.groupby('Género').size()
tabla1 = pd.DataFrame({
    'Frecuencia absoluta': cuenta_variable,
    'Frecuencia relativa (%)': round(cuenta_variable / tamaño * 100, 1)
})
tabla1

Frecuencia absoluta  Frecuencia relativa (%)
Género                                                 
Femenino                   505                     50.5
Masculino                  495                     49.5

**Respuesta**:

El género femenino es el más frecuente y corresponde al 50,5%.

4. ¿Qué jornada tiene mayor representación en la muestra? ¿Hay alguna diferencia relevante entre el porcentaje de estudiantes que pertenece a cada una?

In [ ]:
#Tamaño de la base de datos
tamaño=base_datos.shape[0]

#Tabla de frecuencia variable cualitativa
cuenta_variable = base_datos.groupby('Jornada').size()
tabla2 = pd.DataFrame({
    'Frecuencia absoluta': cuenta_variable,
    'Frecuencia relativa (%)': round(cuenta_variable / tamaño * 100, 1)
})
tabla2

Frecuencia absoluta  Frecuencia relativa (%)
Jornada                                                 
Diuna                       649                     64.9
Vespertina                  351                     35.1

**Respuesta:**

La jornada diurna es la más frecuente. La diferencia es relevante, pues el 64,9% de los alumnos pertenecen a la diurna mientras que el 35,1% pertenecen a la vespertina

5. Realiza una tabla de frecuencias para la variable Asignaturas reprobadas. La tabla debe mostrar las frecuencias puntuales y acumuladas, luego responde las siguientes preguntas:

In [ ]:
#Tamaño de la base de datos
tamaño=base_datos.shape[0]

#Tabla de frecuencia variable cualitativa
cuenta_variable = base_datos.groupby('Asignaturas reprobadas').size()
tabla2 = pd.DataFrame({
    'Frecuencia absoluta (fi)': cuenta_variable,
    'Frecuencia absoluta acumulada (Fi)': cuenta_variable.cumsum(),
    'Frecuencia relativa (%) (hi)': round(cuenta_variable / tamaño * 100, 1),
    'Frecuencia relativa acumulada (%) (Hi)': (cuenta_variable/tamaño * 100).cumsum(),
})
tabla2

Frecuencia absoluta (fi)  \
Asignaturas reprobadas                             
0                                             81   
1                                            200   
2                                            211   
3                                            199   
4                                            209   
5 o más                                      100   

                        Frecuencia absoluta acumulada (Fi)  \
Asignaturas reprobadas                                       
0                                                       81   
1                                                      281   
2                                                      492   
3                                                      691   
4                                                      900   
5 o más                                               1000   

                        Frecuencia relativa (%) (hi)  \
Asignaturas reprobadas                                 
0                                                8.1   
1                                               20.0   
2                                               21.1   
3                                               19.9   
4                                               20.9   
5 o más                                         10.0   

                        Frecuencia relativa acumulada (%) (Hi)  
Asignaturas reprobadas                                          
0                                                          8.1  
1                                                         28.1  
2                                                         49.2  
3                                                         69.1  
4                                                         90.0  
5 o más                                                  100.0

  $\hspace{1cm}$ 5.1 ¿Cuál es la cantidad más común de asignaturas reprobadas?

**Respuesta:**

La cantidad más común es 2 asignaturas con 211 alumnos (21,1%)

$\hspace{1cm}$ 5.2. ¿Qué porcentaje de estudiantes ha reprobado 3 asignaturas o menos?

**Respuesta:**

El 69,1% de los alumnos ha reprobado 3 asignaturas o menos

$\hspace{1cm}$ 5.3. Interpreta las siguientes frecuencias:

$\hspace{1cm}$ $F_{3}$:

$\hspace{1cm}$ $f_{1}$:

$\hspace{1cm}$ $H_{5}$:

$\hspace{1cm}$ $h_{2}$:

## Actividad 3

La base de datos utilizada en esta actividad corresponde al Informe Mundial de la Felicidad publicado por la ONU del año 2024. Contiene información de 140 países y evalúa el nivel de felicidad de sus habitantes en una **escala continua (de 0 a 10)**, así como otros indicadores como el PIB per cápita, apoyo social, esperanza de vida saludable, percepción de corrupción y libertad para tomar decisiones de vida.

Para responder las siguientes preguntas, descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1bKzL1MFdbfXMy9aYpGNt-SXlTrxEEL_9
">aquí</a>**.

*Log PIB per cápita, corresponde al logaritmo del Producto Interno Bruto per cápita. Esta transformación se usa para reducir la asimetría de la distribución del PIB per cápita y facilitar comparaciones cuando hay grandes diferencias entre países.*


1. Define la población y muestra del estudio.

**Respuesta**:

Población: Países del mundo

Muestra: 140 países del mundo

2. ¿Cuál es la clasificación de la variable *Puntaje de felicidad*?

**Respuesta:** Cuantitativa continua

3. Crea una tabla de frecuencias puntuales y acumuladas utilizando la variable *Puntaje de felicidad*. Considera crear la tabla de frecuencias con 8 intervalos.

In [ ]:
import pandas as pd
base_datos= pd.read_csv('Reporte_felicidad_2024.csv')

In [ ]:
base_datos.describe()

Puntaje de felicidad  Log PIB per cápita  Apoyo social  \
count            140.000000          140.000000    140.000000   
mean               5.530893            1.378807      1.134329   
std                1.181247            0.425098      0.333317   
min                1.721000            0.000000      0.000000   
25%                4.631750            1.077750      0.921750   
50%                5.800500            1.431500      1.237500   
75%                6.426250            1.741500      1.383250   
max                7.741000            2.141000      1.617000   

       Esperanza de vida saludable  Libertad para tomar decisiones de vida,  \
count                   140.000000                               140.000000   
mean                      0.520886                                 0.620621   
std                       0.164923                                 0.162492   
min                       0.000000                                 0.000000   
25%                       0.398000                                 0.527500   
50%                       0.549500                                 0.641000   
75%                       0.648500                                 0.736000   
max                       0.857000                                 0.863000   

       Generosidad  Percepciones de corrupción  Distopía + residual  
count   140.000000                  140.000000           140.000000  
mean      0.146271                    0.154121             1.575914  
std       0.073441                    0.126238             0.537459  
min       0.000000                    0.000000            -0.073000  
25%       0.091000                    0.068750             1.308250  
50%       0.136500                    0.120500             1.644500  
75%       0.192500                    0.193750             1.881750  
max       0.401000                    0.575000             2.998000

In [ ]:
base_datos.head(1)

Nombre del país Indicador regional  Puntaje de felicidad  \
0       Finlandia  Europa occidental                 7.741   

   Log PIB per cápita  Apoyo social  Esperanza de vida saludable  \
0               1.844         1.572                        0.695   

   Libertad para tomar decisiones de vida,  Generosidad  \
0                                    0.859        0.142   

   Percepciones de corrupción  Distopía + residual  
0                       0.546                2.082

In [ ]:
#Tamaño de la base de datos
import pandas as pd

tamaño=base_datos.shape[0]

base_datos['Intervalos_variable'] = pd.cut(base_datos['Puntaje de felicidad'], bins = 8, include_lowest= True, precision = 0)

cuenta_intervalos_variable= base_datos.groupby('Intervalos_variable', observed = True).size()

tabla= pd.DataFrame({
    'Frecuencia absoluta (fi)': cuenta_intervalos_variable,
    'Frecuencia absoluta acumulada (Fi)': cuenta_intervalos_variable.cumsum(),
    'Frecuencia relativa (%) (hi)': cuenta_intervalos_variable/tamaño * 100,
    'Frecuencia relativa acumulada (%) (Hi)': (cuenta_intervalos_variable/tamaño * 100).cumsum()
})
round(tabla, 1)

Frecuencia absoluta (fi)  \
Intervalos_variable                                   
(1.5999999999999999, 2.5]                         1   
(2.5, 3.2]                                        2   
(3.2, 4.0]                                       14   
(4.0, 4.7]                                       19   
(4.7, 5.5]                                       25   
(5.5, 6.2]                                       34   
(6.2, 7.0]                                       34   
(7.0, 7.7]                                       11   

                           Frecuencia absoluta acumulada (Fi)  \
Intervalos_variable                                             
(1.5999999999999999, 2.5]                                   1   
(2.5, 3.2]                                                  3   
(3.2, 4.0]                                                 17   
(4.0, 4.7]                                                 36   
(4.7, 5.5]                                                 61   
(5.5, 6.2]                                                 95   
(6.2, 7.0]                                                129   
(7.0, 7.7]                                                140   

                           Frecuencia relativa (%) (hi)  \
Intervalos_variable                                       
(1.5999999999999999, 2.5]                           0.7   
(2.5, 3.2]                                          1.4   
(3.2, 4.0]                                         10.0   
(4.0, 4.7]                                         13.6   
(4.7, 5.5]                                         17.9   
(5.5, 6.2]                                         24.3   
(6.2, 7.0]                                         24.3   
(7.0, 7.7]                                          7.9   

                           Frecuencia relativa acumulada (%) (Hi)  
Intervalos_variable                                                
(1.5999999999999999, 2.5]                                     0.7  
(2.5, 3.2]                                                    2.1  
(3.2, 4.0]                                                   12.1  
(4.0, 4.7]                                                   25.7  
(4.7, 5.5]                                                   43.6  
(5.5, 6.2]                                                   67.9  
(6.2, 7.0]                                                   92.1  
(7.0, 7.7]                                                  100.0

4. Interpreta las siguientes frecuencias:

$\hspace{1cm}$ $f_{3}$: 14 países tienen un índice de felicidad entre 3,2 y 4 puntos

$\hspace{1cm}$ $F_{5}$: 61 países tienen un índice de felicidad de a lo más 5,5 puntos.

$\hspace{1cm}$ $h_{4}$: El 13,6% de los países tienen un indice de felicidad entre 4 y 4,7 puntos

$\hspace{1cm}$ $H_{6}$: El 67,9% de los paísers tiene un índice de felicidad de a lo más 6,2 puntos.

5. ¿Qué puedes concluir sobre la concentración de países según su nivel de felicidad? Para responder considera las siguientes categorías de nivel de felicidad:

* Baja: (1,70; 3,23]
* Media: (3,23; 4,73]
* Media-alta: (4,73; 6,24]
* Alta: (6,24; 7,74]

**Respuesta:**

La mayor frecuencias de países (34) están en la categoría media-alta y alta.

Solo un país está en la categoría baja.

6. ¿Cómo se distribuyen los países según su PIB per cápita? Construye una tabla de frecuencias considerando 8 intervalos.

In [ ]:
#Tamaño de la base de datos
import pandas as pd

tamaño=base_datos.shape[0]

base_datos['Intervalos_variable'] = pd.cut(base_datos['Log PIB per cápita'], bins = 8, include_lowest= True, precision = 0)

cuenta_intervalos_variable= base_datos.groupby('Intervalos_variable', observed = True).size()

tabla2= pd.DataFrame({
    'Frecuencia absoluta (fi)': cuenta_intervalos_variable,
    'Frecuencia absoluta acumulada (Fi)': cuenta_intervalos_variable.cumsum(),
    'Frecuencia relativa (%) (hi)': cuenta_intervalos_variable/tamaño * 100,
    'Frecuencia relativa acumulada (%) (Hi)': (cuenta_intervalos_variable/tamaño * 100).cumsum()
})
round(tabla2, 1)

Frecuencia absoluta (fi)  \
Intervalos_variable                                     
(-0.10200000000000001, 0.3]                         1   
(0.3, 0.5]                                          1   
(0.5, 0.8]                                         17   
(0.8, 1.1]                                         15   
(1.1, 1.3]                                         23   
(1.3, 1.6]                                         33   
(1.6, 1.9]                                         37   
(1.9, 2.1]                                         13   

                             Frecuencia absoluta acumulada (Fi)  \
Intervalos_variable                                               
(-0.10200000000000001, 0.3]                                   1   
(0.3, 0.5]                                                    2   
(0.5, 0.8]                                                   19   
(0.8, 1.1]                                                   34   
(1.1, 1.3]                                                   57   
(1.3, 1.6]                                                   90   
(1.6, 1.9]                                                  127   
(1.9, 2.1]                                                  140   

                             Frecuencia relativa (%) (hi)  \
Intervalos_variable                                         
(-0.10200000000000001, 0.3]                           0.7   
(0.3, 0.5]                                            0.7   
(0.5, 0.8]                                           12.1   
(0.8, 1.1]                                           10.7   
(1.1, 1.3]                                           16.4   
(1.3, 1.6]                                           23.6   
(1.6, 1.9]                                           26.4   
(1.9, 2.1]                                            9.3   

                             Frecuencia relativa acumulada (%) (Hi)  
Intervalos_variable                                                  
(-0.10200000000000001, 0.3]                                     0.7  
(0.3, 0.5]                                                      1.4  
(0.5, 0.8]                                                     13.6  
(0.8, 1.1]                                                     24.3  
(1.1, 1.3]                                                     40.7  
(1.3, 1.6]                                                     64.3  
(1.6, 1.9]                                                     90.7  
(1.9, 2.1]                                                    100.0

7. Considera las siguientes categorías de nivel de PIB per cápita:
* Baja: (-0,0121; 0,54]
* Media: (0,54; 1,07]
* Media-alta: (1,07; 1,61]
* Alta: (1,61; 2,14]

Con base en la tabla de frecuencias del inciso anterior, ¿en qué categoría se concentra la mayoría de los países y qué conclusiones puedes extraer sobre su desarrollo económico?

**Respuesta:** La mayoría de los países se está en la categoría Alta.

8. En el siguiente gráfico se representa la relación entre el logaritmo del PIB per cápita y el puntaje de felicidad, donde cada punto corresponde a un país. ¿Qué ocurre con los países que tienen menor valor de logaritmo del PIB per capita? ¿Y con los que tienen un mayor valor? ¿Observas alguna tendencia?

<img src = "https://i.ibb.co/WNw0PtRf/Captura-de-pantalla-2025-07-22-102447.png" width = "800"/>

## Actividad 4

Para describir un conjunto de llamadas telefónicas se trabajará con una base de datos que contiene información del tiempo de duración y destino de 250  llamadas realizadas a algún país de América del Norte.

Descarga la base de datos **<a href="https://docs.google.com/spreadsheets/d/1KtiWhqgBecg_sKsGur_OCog5a0sU69NQ/export?format=xlsx
">aquí</a>**.


1. Define la población y la muestra del estudio.

2. Muestra las tres primeras filas y tres últimas filas de DataFrame.

3. Clasifica las variables relevantes en el estudio.

4. Construye una tabla de distribución de frecuencias para la variable tiempo de duración de las llamadas considerando frecuencias puntuales y acumuladas. Para la construcción de la tabla considera 9 intervalos.

5. A partir de la pregunta anterior, interpreta las siguientes frecuencias:

$\hspace{1cm}$ $f_{6}$:

$\hspace{1cm}$ $h_{2}$:

$\hspace{1cm}$ $F_{4}$:

$\hspace{1cm}$ $H_{3}$:

## Actividad 5

A continuación, se presentan datos nacionales por comuna de la medición SIMCE de octavo básico rendida en el año 2015.

Descarga la base de datos <a href="https://docs.google.com/spreadsheets/d/1fIGstVw-dQfA5WMq9mvhx3wVOHCXbrpj/export?format=xlsx">aquí</a> y responde las siguientes preguntas:

1. Construye una tabla de distribución de frecuencias (absolutas y porcentuales incluyendo las acumuladas) que resuma los puntajes obtenidos en la prueba SIMCE de Matemática. Considera 9 intervalos para la construcción de la tabla.

$\hspace{1cm}$*Observaciones: Se debe escoger la variable prom_mate8b_com*

In [ ]:
import pandas as pd

#Para cargar archivo excel en Python
excel_file = pd.ExcelFile('SIMCE.xlsx')
sheet_names = excel_file.sheet_names
print(sheet_names)

['Hoja1']


In [ ]:
base_datos=pd.read_excel("SIMCE.xlsx",sheet_name="Hoja1")

In [ ]:
base_datos.head(1)

agno  fecha_bbdd      codigo_bbdd grado  cod_reg  \
0  2015    20160908  final20160908v1    8b       11   

                                     nom_reg  cod_pro nom_pro  cod_com  \
0  AYSÉN DEL GENERAL CARLOS IBÁÑEZ DEL CAMPO        2   AISÉN        1   

  nom_com  ...  nalu_nat8b_com prom_lect8b_com  prom_mate8b_com  \
0   AISÉN  ...             312             237              257   

   prom_nat8b_com  dif_lect8b_com  dif_mate8b_com  dif_nat8b_com  \
0             263             3.0             7.0           -9.0   

   sigdif_lect8b_com  sigdif_mate8b_com  sigdif_nat8b_com  
0   No significativa   No significativa  No significativa  

[1 rows x 24 columns]

In [ ]:
#Tamaño de la base de datos
import pandas as pd

tamaño=base_datos.shape[0]

base_datos['Intervalos_variable'] = pd.cut(base_datos['prom_mate8b_com'], bins = 9, include_lowest= True, precision = 0)

cuenta_intervalos_variable= base_datos.groupby('Intervalos_variable', observed = True).size()

tabla= pd.DataFrame({
    'Frecuencia absoluta (fi)': cuenta_intervalos_variable,
    'Frecuencia absoluta acumulada (Fi)': cuenta_intervalos_variable.cumsum(),
    'Frecuencia relativa (%) (hi)': cuenta_intervalos_variable/tamaño * 100,
    'Frecuencia relativa acumulada (%) (Hi)': (cuenta_intervalos_variable/tamaño * 100).cumsum()
})
round(tabla, 1)

Frecuencia absoluta (fi)  \
Intervalos_variable                             
(193.0, 207.0]                              2   
(207.0, 220.0]                              4   
(220.0, 233.0]                             29   
(233.0, 246.0]                             88   
(246.0, 260.0]                            118   
(260.0, 273.0]                             80   
(273.0, 286.0]                             16   
(286.0, 299.0]                              2   
(299.0, 312.0]                              5   

                     Frecuencia absoluta acumulada (Fi)  \
Intervalos_variable                                       
(193.0, 207.0]                                        2   
(207.0, 220.0]                                        6   
(220.0, 233.0]                                       35   
(233.0, 246.0]                                      123   
(246.0, 260.0]                                      241   
(260.0, 273.0]                                      321   
(273.0, 286.0]                                      337   
(286.0, 299.0]                                      339   
(299.0, 312.0]                                      344   

                     Frecuencia relativa (%) (hi)  \
Intervalos_variable                                 
(193.0, 207.0]                                0.6   
(207.0, 220.0]                                1.2   
(220.0, 233.0]                                8.4   
(233.0, 246.0]                               25.6   
(246.0, 260.0]                               34.3   
(260.0, 273.0]                               23.3   
(273.0, 286.0]                                4.7   
(286.0, 299.0]                                0.6   
(299.0, 312.0]                                1.5   

                     Frecuencia relativa acumulada (%) (Hi)  
Intervalos_variable                                          
(193.0, 207.0]                                          0.6  
(207.0, 220.0]                                          1.7  
(220.0, 233.0]                                         10.2  
(233.0, 246.0]                                         35.8  
(246.0, 260.0]                                         70.1  
(260.0, 273.0]                                         93.3  
(273.0, 286.0]                                         98.0  
(286.0, 299.0]                                         98.5  
(299.0, 312.0]                                        100.0

2. ¿Es correcto afirmar que los resultados de Matemática se concentran en los puntajes más bajos? Justifica tu respuesta.

3.  Diseñe una tabla de frecuencias relativas que resuma la información de la cantidad de comunas por región. Realice observaciones de los datos de la tabla y construya una hipótesis al respecto.